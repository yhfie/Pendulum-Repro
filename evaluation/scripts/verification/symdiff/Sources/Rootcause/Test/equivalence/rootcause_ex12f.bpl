// mem_eq.bpl
//
// Example with equivalence checking 
// comparing the entire memory
//


// Run this through Boogie to see why E fails 
//Debugging E above 
procedure EQ_F(index:int, h:[int]int) {
  var h1:[int]int, h2:[int]int;
  //here is the model
  assume(index == 719 && h[719] == 0); //the model that violates asserttion A1
  //Version 1
  h1 := h;
  h1[index] := 1;
  h1[index+1] := 2;

  //this is the marker that is understood by Rootcause
  goto AA_INSTR_EQ_BODY$1;
  AA_INSTR_EQ_BODY$1: 
   

  //Version 2
  h2 := h;
  h2[index+1] := 2;
  //h2[index] := 1;
  //assert(forall i:int :: h1[i] == h2[i]); //A1
  assert(exists i:int  :: {h1[i]}{h2[i]} h1[i] != h2[i]); //negation of A1
}
