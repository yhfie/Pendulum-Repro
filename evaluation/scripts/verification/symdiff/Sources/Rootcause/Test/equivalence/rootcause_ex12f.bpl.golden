BENCHMARK:rootcause_ex12f.bpl(ERRORFREE)
Processing file rootcause_ex12f.bpl
############# Implementation = EQ_F #################
Cause ==>
leftAssignment: h1 := h;
rightAssignment: h2[index + 1] := 2;
